### DISCUSSION TITLE
Connect to controller via Unifi Portal?

### DISCUSSION THREAD

**Discussion by MarkV74** [Art-of-WiFi/UniFi-API-client#164] [Q&A] [answered]
Hi, i have some clients running on a CKgen2 or UDM. All of them are added to my Unifi Portal with the same Ubiquiti account. Is it possible to use our product directlty on this portal or do i have to create port forwardings to the controllers on every site?

Thanks!
Mark

**Comment by malle-pietje** [comment]
The unifi.ui.com portal communications uses WebRTC which isn’t supported by PHP. For this reason this library doesn’t support unifi.ui.com. There is a Node.js-based library that provides experimental support for unifi.ui.com, the last time I tested it things were quite unreliable…

**Reply by thib3113** [reply]
hum ? WebRTC ? did you have more informations about it ? because, when I go to unifi.ui.com, on my side, it just call the normal API ... just it use a custom domain, and a different login method .

Also, it use websockets for events, but the same way the local environment use it .

**Reply by malle-pietje** [reply]
More info here:
https://github.com/delian/node-unifiapi

**Reply by thib3113** [reply, marked as answer]
Thank you . 

After some search, the REST API is used only if you use the portal inside the LAN of the UDM ... outside it use raw websockets messages .

**Reply by MarkV74** [reply]
Thanks for your reply, i hoped it was easy to accomplish but will consider creating port forwardings on every site.
